### PR TITLE
[google compute] Add License resource

### PR DESCRIPTION
**Pull request by erjohnso**
Adds Licenses[1] support to the GCE driver to further improve full API coverage.

[1] https://cloud.google.com/compute/docs/reference/latest/licenses

### CODE REVIEW

File: libcloud/compute/drivers/gce.py [hunk 1]
```diff
@@ -77,6 +77,21 @@ def __init__(self, user_id, key, secure, auth_type=None,
...
 
 
+class GCELicense(UuidMixin):
+    """A GCE License used to track software usage in GCE nodes."""
+    def __init__(self, id, name, driver, charges_use_fee, extra=None):
+        self.id = str(id)
+        self.name = name
+        self.driver = driver
+        self.charges_use_fee = charges_use_fee
+        self.extra = extra
```
Comment (Kami) [inline libcloud/compute/drivers/gce.py]: `extra = extra or {}`

File: libcloud/compute/drivers/gce.py [hunk 2]
```diff
@@ -4158,7 +4196,13 @@ def _to_node_image(self, image):
         if 'sourceDiskId' in image:
             extra['sourceDiskId'] = image.get('sourceDiskId', None)
         if 'licenses' in image:
-            extra['licenses'] = image.get('licenses', None)
+            extra['licenses'] = []
+            for license in image['licenses']:
+                selfLink_parts = license.split('/')
+                license_project = selfLink_parts[6]
+                license_name = selfLink_parts[-1]
+                extra['licenses'].append(self.ex_get_license(license_project,
```
Comment (Kami) [inline libcloud/compute/drivers/gce.py]: To make code easier to read, follow and update, please prefer using keyword arguments over regular arguments.

File: libcloud/compute/drivers/gce.py [hunk 3]
```diff
@@ -4336,7 +4380,13 @@ def _to_snapshot(self, snapshot):
         if 'storageBytesStatus' in snapshot:
             extra['storageBytesStatus'] = snapshot['storageBytesStatus']
         if 'licenses' in snapshot:
-            extra['licenses'] = snapshot['licenses']
+            extra['licenses'] = []
+            for license in snapshot['licenses']:
```
Comment (Kami) [inline libcloud/compute/drivers/gce.py]: Looks like this chunk of code is repeated in two place, might be worth refactoring it in a re-usable method :)

**Comment by erjohnso** [comment]
Thanks @Kami. Addressed your feedback on this, so I'll go ahead and do the merge upstream once Travis gives the OK.

**Pull request closed without merging**